discovery_engine/lib/src/domain/feed_manager.dart

```diff
@@ -204,4 +205,9 @@ class FeedManager {
     final sources = await _sourcePreferenceRepository.getTrusted();
     return EngineEvent.trustedSourcesListRequestSucceeded(sources);
   }
+
+  Future<EngineEvent> getAvailableSourcesList(String fuzzySearchTerm) async {
+    final sources = await _engine.getAvailableSources(fuzzySearchTerm);
+    return EngineEvent.availableSourcesListRequestSucceeded(sources);
```

Comment:
If we use only this one event, maybe we don't need `AvailableSourcesListRequestFailed`? otherwise maybe we could check if the list is empty, and return it in that case, wdyt?